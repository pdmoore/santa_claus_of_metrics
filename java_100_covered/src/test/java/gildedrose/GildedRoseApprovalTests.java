package gildedrose;

import org.approvaltests.Approvals;
import org.approvaltests.combinations.CombinationApprovals;
import org.approvaltests.reporters.UseReporter;
import org.approvaltests.reporters.macosx.DiffMergeReporter;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;

//Uncomment the following line to use DiffMerge when Received/Approved differ
//@UseReporter(DiffMergeReporter.class)
public class GildedRoseApprovalTests {

    @Test
    public void test_UpdateQuality_SingleItem_AtSellin_NoQuality() {
        // Arrange
        Item[] items = new Item[]{new Item("normal", 0, 0)};
        GildedRose sut = new GildedRose(items);

        // Act
        sut.updateQuality();

        // Assert
        // The Verify call creates a file named GildedRoseApprovalTests.OneItemApproval.received.txt
        // and compares it against the Approved result file, GildedRoseApprovalTests.OneItemApproval.approved.txt
        // If the contents of the two files match, the test passes
        // otherwise the test fails and the diff of the two files is shown
        Approvals.verify(sut.items[0].toString());
    }

    @Test
    public void test_UpdateQuality_SingleItem() {
        CombinationApprovals.verifyAllCombinations(
                this::doUpdateQuality,
                new String[] {"foo", "Aged Brie", "Backstage passes to a TAFKAL80ETC concert", "Sulfuras, Hand of Ragnaros"},
                new Integer[] {-1, 0, 5, 10, 11},    // SellIn
                new Integer[] { 0, 1, 49, 50 }       // Quality
        );
    }

    private String doUpdateQuality(String name, int sellIn, int quality) {
        Item[] items = new Item[]{new Item(name, sellIn, quality)};
        GildedRose sut = new GildedRose(items);
        sut.updateQuality();
        return sut.items[0].toString();
    }
}
