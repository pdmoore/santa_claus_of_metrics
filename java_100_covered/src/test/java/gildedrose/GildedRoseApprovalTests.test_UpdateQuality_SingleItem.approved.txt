[foo, -1, 0] => foo, -2, 0 
[foo, -1, 1] => foo, -2, 0 
[foo, -1, 5] => foo, -2, 3 
[foo, -1, 10] => foo, -2, 8 
[foo, -1, 49] => foo, -2, 47 
[foo, -1, 50] => foo, -2, 48 
[foo, 0, 0] => foo, -1, 0 
[foo, 0, 1] => foo, -1, 0 
[foo, 0, 5] => foo, -1, 3 
[foo, 0, 10] => foo, -1, 8 
[foo, 0, 49] => foo, -1, 47 
[foo, 0, 50] => foo, -1, 48 
[foo, 1, 0] => foo, 0, 0 
[foo, 1, 1] => foo, 0, 0 
[foo, 1, 5] => foo, 0, 4 
[foo, 1, 10] => foo, 0, 9 
[foo, 1, 49] => foo, 0, 48 
[foo, 1, 50] => foo, 0, 49 
[foo, 2, 0] => foo, 1, 0 
[foo, 2, 1] => foo, 1, 0 
[foo, 2, 5] => foo, 1, 4 
[foo, 2, 10] => foo, 1, 9 
[foo, 2, 49] => foo, 1, 48 
[foo, 2, 50] => foo, 1, 49 
[foo, 3, 0] => foo, 2, 0 
[foo, 3, 1] => foo, 2, 0 
[foo, 3, 5] => foo, 2, 4 
[foo, 3, 10] => foo, 2, 9 
[foo, 3, 49] => foo, 2, 48 
[foo, 3, 50] => foo, 2, 49 
[foo, 4, 0] => foo, 3, 0 
[foo, 4, 1] => foo, 3, 0 
[foo, 4, 5] => foo, 3, 4 
[foo, 4, 10] => foo, 3, 9 
[foo, 4, 49] => foo, 3, 48 
[foo, 4, 50] => foo, 3, 49 
[foo, 5, 0] => foo, 4, 0 
[foo, 5, 1] => foo, 4, 0 
[foo, 5, 5] => foo, 4, 4 
[foo, 5, 10] => foo, 4, 9 
[foo, 5, 49] => foo, 4, 48 
[foo, 5, 50] => foo, 4, 49 
[foo, 6, 0] => foo, 5, 0 
[foo, 6, 1] => foo, 5, 0 
[foo, 6, 5] => foo, 5, 4 
[foo, 6, 10] => foo, 5, 9 
[foo, 6, 49] => foo, 5, 48 
[foo, 6, 50] => foo, 5, 49 
[foo, 9, 0] => foo, 8, 0 
[foo, 9, 1] => foo, 8, 0 
[foo, 9, 5] => foo, 8, 4 
[foo, 9, 10] => foo, 8, 9 
[foo, 9, 49] => foo, 8, 48 
[foo, 9, 50] => foo, 8, 49 
[foo, 10, 0] => foo, 9, 0 
[foo, 10, 1] => foo, 9, 0 
[foo, 10, 5] => foo, 9, 4 
[foo, 10, 10] => foo, 9, 9 
[foo, 10, 49] => foo, 9, 48 
[foo, 10, 50] => foo, 9, 49 
[foo, 11, 0] => foo, 10, 0 
[foo, 11, 1] => foo, 10, 0 
[foo, 11, 5] => foo, 10, 4 
[foo, 11, 10] => foo, 10, 9 
[foo, 11, 49] => foo, 10, 48 
[foo, 11, 50] => foo, 10, 49 
[Aged Brie, -1, 0] => Aged Brie, -2, 2 
[Aged Brie, -1, 1] => Aged Brie, -2, 3 
[Aged Brie, -1, 5] => Aged Brie, -2, 7 
[Aged Brie, -1, 10] => Aged Brie, -2, 12 
[Aged Brie, -1, 49] => Aged Brie, -2, 50 
[Aged Brie, -1, 50] => Aged Brie, -2, 50 
[Aged Brie, 0, 0] => Aged Brie, -1, 2 
[Aged Brie, 0, 1] => Aged Brie, -1, 3 
[Aged Brie, 0, 5] => Aged Brie, -1, 7 
[Aged Brie, 0, 10] => Aged Brie, -1, 12 
[Aged Brie, 0, 49] => Aged Brie, -1, 50 
[Aged Brie, 0, 50] => Aged Brie, -1, 50 
[Aged Brie, 1, 0] => Aged Brie, 0, 1 
[Aged Brie, 1, 1] => Aged Brie, 0, 2 
[Aged Brie, 1, 5] => Aged Brie, 0, 6 
[Aged Brie, 1, 10] => Aged Brie, 0, 11 
[Aged Brie, 1, 49] => Aged Brie, 0, 50 
[Aged Brie, 1, 50] => Aged Brie, 0, 50 
[Aged Brie, 2, 0] => Aged Brie, 1, 1 
[Aged Brie, 2, 1] => Aged Brie, 1, 2 
[Aged Brie, 2, 5] => Aged Brie, 1, 6 
[Aged Brie, 2, 10] => Aged Brie, 1, 11 
[Aged Brie, 2, 49] => Aged Brie, 1, 50 
[Aged Brie, 2, 50] => Aged Brie, 1, 50 
[Aged Brie, 3, 0] => Aged Brie, 2, 1 
[Aged Brie, 3, 1] => Aged Brie, 2, 2 
[Aged Brie, 3, 5] => Aged Brie, 2, 6 
[Aged Brie, 3, 10] => Aged Brie, 2, 11 
[Aged Brie, 3, 49] => Aged Brie, 2, 50 
[Aged Brie, 3, 50] => Aged Brie, 2, 50 
[Aged Brie, 4, 0] => Aged Brie, 3, 1 
[Aged Brie, 4, 1] => Aged Brie, 3, 2 
[Aged Brie, 4, 5] => Aged Brie, 3, 6 
[Aged Brie, 4, 10] => Aged Brie, 3, 11 
[Aged Brie, 4, 49] => Aged Brie, 3, 50 
[Aged Brie, 4, 50] => Aged Brie, 3, 50 
[Aged Brie, 5, 0] => Aged Brie, 4, 1 
[Aged Brie, 5, 1] => Aged Brie, 4, 2 
[Aged Brie, 5, 5] => Aged Brie, 4, 6 
[Aged Brie, 5, 10] => Aged Brie, 4, 11 
[Aged Brie, 5, 49] => Aged Brie, 4, 50 
[Aged Brie, 5, 50] => Aged Brie, 4, 50 
[Aged Brie, 6, 0] => Aged Brie, 5, 1 
[Aged Brie, 6, 1] => Aged Brie, 5, 2 
[Aged Brie, 6, 5] => Aged Brie, 5, 6 
[Aged Brie, 6, 10] => Aged Brie, 5, 11 
[Aged Brie, 6, 49] => Aged Brie, 5, 50 
[Aged Brie, 6, 50] => Aged Brie, 5, 50 
[Aged Brie, 9, 0] => Aged Brie, 8, 1 
[Aged Brie, 9, 1] => Aged Brie, 8, 2 
[Aged Brie, 9, 5] => Aged Brie, 8, 6 
[Aged Brie, 9, 10] => Aged Brie, 8, 11 
[Aged Brie, 9, 49] => Aged Brie, 8, 50 
[Aged Brie, 9, 50] => Aged Brie, 8, 50 
[Aged Brie, 10, 0] => Aged Brie, 9, 1 
[Aged Brie, 10, 1] => Aged Brie, 9, 2 
[Aged Brie, 10, 5] => Aged Brie, 9, 6 
[Aged Brie, 10, 10] => Aged Brie, 9, 11 
[Aged Brie, 10, 49] => Aged Brie, 9, 50 
[Aged Brie, 10, 50] => Aged Brie, 9, 50 
[Aged Brie, 11, 0] => Aged Brie, 10, 1 
[Aged Brie, 11, 1] => Aged Brie, 10, 2 
[Aged Brie, 11, 5] => Aged Brie, 10, 6 
[Aged Brie, 11, 10] => Aged Brie, 10, 11 
[Aged Brie, 11, 49] => Aged Brie, 10, 50 
[Aged Brie, 11, 50] => Aged Brie, 10, 50 
[Backstage passes to a TAFKAL80ETC concert, -1, 0] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, -1, 1] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, -1, 5] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, -1, 10] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, -1, 49] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, -1, 50] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 0] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 1] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 5] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 10] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 49] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 50] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 1, 0] => Backstage passes to a TAFKAL80ETC concert, 0, 3 
[Backstage passes to a TAFKAL80ETC concert, 1, 1] => Backstage passes to a TAFKAL80ETC concert, 0, 4 
[Backstage passes to a TAFKAL80ETC concert, 1, 5] => Backstage passes to a TAFKAL80ETC concert, 0, 8 
[Backstage passes to a TAFKAL80ETC concert, 1, 10] => Backstage passes to a TAFKAL80ETC concert, 0, 13 
[Backstage passes to a TAFKAL80ETC concert, 1, 49] => Backstage passes to a TAFKAL80ETC concert, 0, 50 
[Backstage passes to a TAFKAL80ETC concert, 1, 50] => Backstage passes to a TAFKAL80ETC concert, 0, 50 
[Backstage passes to a TAFKAL80ETC concert, 2, 0] => Backstage passes to a TAFKAL80ETC concert, 1, 3 
[Backstage passes to a TAFKAL80ETC concert, 2, 1] => Backstage passes to a TAFKAL80ETC concert, 1, 4 
[Backstage passes to a TAFKAL80ETC concert, 2, 5] => Backstage passes to a TAFKAL80ETC concert, 1, 8 
[Backstage passes to a TAFKAL80ETC concert, 2, 10] => Backstage passes to a TAFKAL80ETC concert, 1, 13 
[Backstage passes to a TAFKAL80ETC concert, 2, 49] => Backstage passes to a TAFKAL80ETC concert, 1, 50 
[Backstage passes to a TAFKAL80ETC concert, 2, 50] => Backstage passes to a TAFKAL80ETC concert, 1, 50 
[Backstage passes to a TAFKAL80ETC concert, 3, 0] => Backstage passes to a TAFKAL80ETC concert, 2, 3 
[Backstage passes to a TAFKAL80ETC concert, 3, 1] => Backstage passes to a TAFKAL80ETC concert, 2, 4 
[Backstage passes to a TAFKAL80ETC concert, 3, 5] => Backstage passes to a TAFKAL80ETC concert, 2, 8 
[Backstage passes to a TAFKAL80ETC concert, 3, 10] => Backstage passes to a TAFKAL80ETC concert, 2, 13 
[Backstage passes to a TAFKAL80ETC concert, 3, 49] => Backstage passes to a TAFKAL80ETC concert, 2, 50 
[Backstage passes to a TAFKAL80ETC concert, 3, 50] => Backstage passes to a TAFKAL80ETC concert, 2, 50 
[Backstage passes to a TAFKAL80ETC concert, 4, 0] => Backstage passes to a TAFKAL80ETC concert, 3, 3 
[Backstage passes to a TAFKAL80ETC concert, 4, 1] => Backstage passes to a TAFKAL80ETC concert, 3, 4 
[Backstage passes to a TAFKAL80ETC concert, 4, 5] => Backstage passes to a TAFKAL80ETC concert, 3, 8 
[Backstage passes to a TAFKAL80ETC concert, 4, 10] => Backstage passes to a TAFKAL80ETC concert, 3, 13 
[Backstage passes to a TAFKAL80ETC concert, 4, 49] => Backstage passes to a TAFKAL80ETC concert, 3, 50 
[Backstage passes to a TAFKAL80ETC concert, 4, 50] => Backstage passes to a TAFKAL80ETC concert, 3, 50 
[Backstage passes to a TAFKAL80ETC concert, 5, 0] => Backstage passes to a TAFKAL80ETC concert, 4, 3 
[Backstage passes to a TAFKAL80ETC concert, 5, 1] => Backstage passes to a TAFKAL80ETC concert, 4, 4 
[Backstage passes to a TAFKAL80ETC concert, 5, 5] => Backstage passes to a TAFKAL80ETC concert, 4, 8 
[Backstage passes to a TAFKAL80ETC concert, 5, 10] => Backstage passes to a TAFKAL80ETC concert, 4, 13 
[Backstage passes to a TAFKAL80ETC concert, 5, 49] => Backstage passes to a TAFKAL80ETC concert, 4, 50 
[Backstage passes to a TAFKAL80ETC concert, 5, 50] => Backstage passes to a TAFKAL80ETC concert, 4, 50 
[Backstage passes to a TAFKAL80ETC concert, 6, 0] => Backstage passes to a TAFKAL80ETC concert, 5, 2 
[Backstage passes to a TAFKAL80ETC concert, 6, 1] => Backstage passes to a TAFKAL80ETC concert, 5, 3 
[Backstage passes to a TAFKAL80ETC concert, 6, 5] => Backstage passes to a TAFKAL80ETC concert, 5, 7 
[Backstage passes to a TAFKAL80ETC concert, 6, 10] => Backstage passes to a TAFKAL80ETC concert, 5, 12 
[Backstage passes to a TAFKAL80ETC concert, 6, 49] => Backstage passes to a TAFKAL80ETC concert, 5, 50 
[Backstage passes to a TAFKAL80ETC concert, 6, 50] => Backstage passes to a TAFKAL80ETC concert, 5, 50 
[Backstage passes to a TAFKAL80ETC concert, 9, 0] => Backstage passes to a TAFKAL80ETC concert, 8, 2 
[Backstage passes to a TAFKAL80ETC concert, 9, 1] => Backstage passes to a TAFKAL80ETC concert, 8, 3 
[Backstage passes to a TAFKAL80ETC concert, 9, 5] => Backstage passes to a TAFKAL80ETC concert, 8, 7 
[Backstage passes to a TAFKAL80ETC concert, 9, 10] => Backstage passes to a TAFKAL80ETC concert, 8, 12 
[Backstage passes to a TAFKAL80ETC concert, 9, 49] => Backstage passes to a TAFKAL80ETC concert, 8, 50 
[Backstage passes to a TAFKAL80ETC concert, 9, 50] => Backstage passes to a TAFKAL80ETC concert, 8, 50 
[Backstage passes to a TAFKAL80ETC concert, 10, 0] => Backstage passes to a TAFKAL80ETC concert, 9, 2 
[Backstage passes to a TAFKAL80ETC concert, 10, 1] => Backstage passes to a TAFKAL80ETC concert, 9, 3 
[Backstage passes to a TAFKAL80ETC concert, 10, 5] => Backstage passes to a TAFKAL80ETC concert, 9, 7 
[Backstage passes to a TAFKAL80ETC concert, 10, 10] => Backstage passes to a TAFKAL80ETC concert, 9, 12 
[Backstage passes to a TAFKAL80ETC concert, 10, 49] => Backstage passes to a TAFKAL80ETC concert, 9, 50 
[Backstage passes to a TAFKAL80ETC concert, 10, 50] => Backstage passes to a TAFKAL80ETC concert, 9, 50 
[Backstage passes to a TAFKAL80ETC concert, 11, 0] => Backstage passes to a TAFKAL80ETC concert, 10, 1 
[Backstage passes to a TAFKAL80ETC concert, 11, 1] => Backstage passes to a TAFKAL80ETC concert, 10, 2 
[Backstage passes to a TAFKAL80ETC concert, 11, 5] => Backstage passes to a TAFKAL80ETC concert, 10, 6 
[Backstage passes to a TAFKAL80ETC concert, 11, 10] => Backstage passes to a TAFKAL80ETC concert, 10, 11 
[Backstage passes to a TAFKAL80ETC concert, 11, 49] => Backstage passes to a TAFKAL80ETC concert, 10, 50 
[Backstage passes to a TAFKAL80ETC concert, 11, 50] => Backstage passes to a TAFKAL80ETC concert, 10, 50 
[Sulfuras, Hand of Ragnaros, -1, 0] => Sulfuras, Hand of Ragnaros, -1, 0 
[Sulfuras, Hand of Ragnaros, -1, 1] => Sulfuras, Hand of Ragnaros, -1, 1 
[Sulfuras, Hand of Ragnaros, -1, 5] => Sulfuras, Hand of Ragnaros, -1, 5 
[Sulfuras, Hand of Ragnaros, -1, 10] => Sulfuras, Hand of Ragnaros, -1, 10 
[Sulfuras, Hand of Ragnaros, -1, 49] => Sulfuras, Hand of Ragnaros, -1, 49 
[Sulfuras, Hand of Ragnaros, -1, 50] => Sulfuras, Hand of Ragnaros, -1, 50 
[Sulfuras, Hand of Ragnaros, 0, 0] => Sulfuras, Hand of Ragnaros, 0, 0 
[Sulfuras, Hand of Ragnaros, 0, 1] => Sulfuras, Hand of Ragnaros, 0, 1 
[Sulfuras, Hand of Ragnaros, 0, 5] => Sulfuras, Hand of Ragnaros, 0, 5 
[Sulfuras, Hand of Ragnaros, 0, 10] => Sulfuras, Hand of Ragnaros, 0, 10 
[Sulfuras, Hand of Ragnaros, 0, 49] => Sulfuras, Hand of Ragnaros, 0, 49 
[Sulfuras, Hand of Ragnaros, 0, 50] => Sulfuras, Hand of Ragnaros, 0, 50 
[Sulfuras, Hand of Ragnaros, 1, 0] => Sulfuras, Hand of Ragnaros, 1, 0 
[Sulfuras, Hand of Ragnaros, 1, 1] => Sulfuras, Hand of Ragnaros, 1, 1 
[Sulfuras, Hand of Ragnaros, 1, 5] => Sulfuras, Hand of Ragnaros, 1, 5 
[Sulfuras, Hand of Ragnaros, 1, 10] => Sulfuras, Hand of Ragnaros, 1, 10 
[Sulfuras, Hand of Ragnaros, 1, 49] => Sulfuras, Hand of Ragnaros, 1, 49 
[Sulfuras, Hand of Ragnaros, 1, 50] => Sulfuras, Hand of Ragnaros, 1, 50 
[Sulfuras, Hand of Ragnaros, 2, 0] => Sulfuras, Hand of Ragnaros, 2, 0 
[Sulfuras, Hand of Ragnaros, 2, 1] => Sulfuras, Hand of Ragnaros, 2, 1 
[Sulfuras, Hand of Ragnaros, 2, 5] => Sulfuras, Hand of Ragnaros, 2, 5 
[Sulfuras, Hand of Ragnaros, 2, 10] => Sulfuras, Hand of Ragnaros, 2, 10 
[Sulfuras, Hand of Ragnaros, 2, 49] => Sulfuras, Hand of Ragnaros, 2, 49 
[Sulfuras, Hand of Ragnaros, 2, 50] => Sulfuras, Hand of Ragnaros, 2, 50 
[Sulfuras, Hand of Ragnaros, 3, 0] => Sulfuras, Hand of Ragnaros, 3, 0 
[Sulfuras, Hand of Ragnaros, 3, 1] => Sulfuras, Hand of Ragnaros, 3, 1 
[Sulfuras, Hand of Ragnaros, 3, 5] => Sulfuras, Hand of Ragnaros, 3, 5 
[Sulfuras, Hand of Ragnaros, 3, 10] => Sulfuras, Hand of Ragnaros, 3, 10 
[Sulfuras, Hand of Ragnaros, 3, 49] => Sulfuras, Hand of Ragnaros, 3, 49 
[Sulfuras, Hand of Ragnaros, 3, 50] => Sulfuras, Hand of Ragnaros, 3, 50 
[Sulfuras, Hand of Ragnaros, 4, 0] => Sulfuras, Hand of Ragnaros, 4, 0 
[Sulfuras, Hand of Ragnaros, 4, 1] => Sulfuras, Hand of Ragnaros, 4, 1 
[Sulfuras, Hand of Ragnaros, 4, 5] => Sulfuras, Hand of Ragnaros, 4, 5 
[Sulfuras, Hand of Ragnaros, 4, 10] => Sulfuras, Hand of Ragnaros, 4, 10 
[Sulfuras, Hand of Ragnaros, 4, 49] => Sulfuras, Hand of Ragnaros, 4, 49 
[Sulfuras, Hand of Ragnaros, 4, 50] => Sulfuras, Hand of Ragnaros, 4, 50 
[Sulfuras, Hand of Ragnaros, 5, 0] => Sulfuras, Hand of Ragnaros, 5, 0 
[Sulfuras, Hand of Ragnaros, 5, 1] => Sulfuras, Hand of Ragnaros, 5, 1 
[Sulfuras, Hand of Ragnaros, 5, 5] => Sulfuras, Hand of Ragnaros, 5, 5 
[Sulfuras, Hand of Ragnaros, 5, 10] => Sulfuras, Hand of Ragnaros, 5, 10 
[Sulfuras, Hand of Ragnaros, 5, 49] => Sulfuras, Hand of Ragnaros, 5, 49 
[Sulfuras, Hand of Ragnaros, 5, 50] => Sulfuras, Hand of Ragnaros, 5, 50 
[Sulfuras, Hand of Ragnaros, 6, 0] => Sulfuras, Hand of Ragnaros, 6, 0 
[Sulfuras, Hand of Ragnaros, 6, 1] => Sulfuras, Hand of Ragnaros, 6, 1 
[Sulfuras, Hand of Ragnaros, 6, 5] => Sulfuras, Hand of Ragnaros, 6, 5 
[Sulfuras, Hand of Ragnaros, 6, 10] => Sulfuras, Hand of Ragnaros, 6, 10 
[Sulfuras, Hand of Ragnaros, 6, 49] => Sulfuras, Hand of Ragnaros, 6, 49 
[Sulfuras, Hand of Ragnaros, 6, 50] => Sulfuras, Hand of Ragnaros, 6, 50 
[Sulfuras, Hand of Ragnaros, 9, 0] => Sulfuras, Hand of Ragnaros, 9, 0 
[Sulfuras, Hand of Ragnaros, 9, 1] => Sulfuras, Hand of Ragnaros, 9, 1 
[Sulfuras, Hand of Ragnaros, 9, 5] => Sulfuras, Hand of Ragnaros, 9, 5 
[Sulfuras, Hand of Ragnaros, 9, 10] => Sulfuras, Hand of Ragnaros, 9, 10 
[Sulfuras, Hand of Ragnaros, 9, 49] => Sulfuras, Hand of Ragnaros, 9, 49 
[Sulfuras, Hand of Ragnaros, 9, 50] => Sulfuras, Hand of Ragnaros, 9, 50 
[Sulfuras, Hand of Ragnaros, 10, 0] => Sulfuras, Hand of Ragnaros, 10, 0 
[Sulfuras, Hand of Ragnaros, 10, 1] => Sulfuras, Hand of Ragnaros, 10, 1 
[Sulfuras, Hand of Ragnaros, 10, 5] => Sulfuras, Hand of Ragnaros, 10, 5 
[Sulfuras, Hand of Ragnaros, 10, 10] => Sulfuras, Hand of Ragnaros, 10, 10 
[Sulfuras, Hand of Ragnaros, 10, 49] => Sulfuras, Hand of Ragnaros, 10, 49 
[Sulfuras, Hand of Ragnaros, 10, 50] => Sulfuras, Hand of Ragnaros, 10, 50 
[Sulfuras, Hand of Ragnaros, 11, 0] => Sulfuras, Hand of Ragnaros, 11, 0 
[Sulfuras, Hand of Ragnaros, 11, 1] => Sulfuras, Hand of Ragnaros, 11, 1 
[Sulfuras, Hand of Ragnaros, 11, 5] => Sulfuras, Hand of Ragnaros, 11, 5 
[Sulfuras, Hand of Ragnaros, 11, 10] => Sulfuras, Hand of Ragnaros, 11, 10 
[Sulfuras, Hand of Ragnaros, 11, 49] => Sulfuras, Hand of Ragnaros, 11, 49 
[Sulfuras, Hand of Ragnaros, 11, 50] => Sulfuras, Hand of Ragnaros, 11, 50 
